src/libcore/fmt/builders.rs

```diff
@@ -124,6 +149,30 @@ impl<'a, 'b: 'a> DebugStruct<'a, 'b> {
     }
 
     /// Finishes output and returns any error encountered.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fmt;
+    ///
+    /// struct Bar {
+    ///     bar: i32,
+    ///     baz: String,
+    /// }
+    ///
+    /// impl fmt::Debug for Bar {
+    ///     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+    ///         fmt.debug_struct("Bar")
+    ///            .field("bar", &self.bar)
+    ///            .field("baz", &self.baz)
+    ///            .finish() // You need to call it in order to "finish" the
+    ///                      // struct formatting.
+    ///     }
+    /// }
+    ///
+    /// // prints "Bar { bar: 10, baz: "Hello World", not_existing_field: 1 }"
```

Comment:
has `not_existing_field` in it where the debug code doesn't

Reply:
Good catch, thanks!